### ISSUE TITLE
Cannot connect to document db through java api (EDIT: solved, but there must be something missing in the tutorial)

### ISSUE THREAD

**Issue by federicovindigni** [orientechnologies/orientdb#6108]
Hi everybody,

I hope someone can help me. I am new to OrientDB and I am trying to get familiar with java api. As first application I tried a simple connection to a test database. It doesn't matter what I do, I get always the same error. I have no problem access the db from the console or the studio, but through java noway.
I tried to change db, to add a new user, nothing.

I'm usign eclipse (yes I've add orient-db-client, orient-db-common and orient-db-enterprise jars), orientdb version is 2.1.15, jdk version is 1.8.0_60

Here is the code

`package test1;

import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;

public class DBapp1 {

	public static void main(String[] args) {
		ODatabaseDocumentTx db = new ODatabaseDocumentTx("remote:localhost/testdb");
		db.open("testu","xxx");
		db.close();
	}	
}`

And this is the error I get

`Exception in thread "main" java.lang.NoClassDefFoundError: com/googlecode/concurrentlinkedhashmap/ConcurrentLinkedHashMap$Builder
	at com.orientechnologies.common.profiler.OProfilerStub.<init>(OProfilerStub.java:38)
	at com.orientechnologies.orient.core.Orient.startup(Orient.java:199)
	at com.orientechnologies.orient.core.Orient.<clinit>(Orient.java:87)
	at com.orientechnologies.orient.core.serialization.serializer.record.string.ORecordSerializerStringAbstract.<clinit>(ORecordSerializerStringAbstract.java:52)
	at com.orientechnologies.orient.core.serialization.serializer.record.ORecordSerializerFactory.<init>(ORecordSerializerFactory.java:47)
	at com.orientechnologies.orient.core.serialization.serializer.record.ORecordSerializerFactory.<clinit>(ORecordSerializerFactory.java:37)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.<clinit>(ODatabaseDocumentTx.java:130)
	at test1.DBapp1.main(DBapp1.java:8)
Caused by: java.lang.ClassNotFoundException: com.googlecode.concurrentlinkedhashmap.ConcurrentLinkedHashMap$Builder
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 8 more
`

EDIT : I found out the problem. 

It's about the imported external jars: the tutorial says ( [http://orientdb.com/docs/last/Document-Database.html](url) ) that I need t add orient-core-*.jar (plus orient-client-*.jar and orient-enterprise-*.jar for remote db) to my classpath. Doing so I get the problem mentioned above.
Right now, only for try, I've included all jars in $ORIENT_HOME/lib directory and the application runs correctly (I've also succesfully listed all the records of class).

So: which jar(s) is missing on the tutorial?

**Comment by tglman** [comment]
hi @federicovindigni,

how do you manage the dependencies ? it seems you miss the concurrentlinkedhashmap-lru jar, that is in the maven dependencises of orientdb-core.

Regards

**Comment by federicovindigni** [comment]
Thanks for the reply.
Sincerely I don't know about concurrentlinkedhashmap-lru. My development enviroment is very straightforward: I downloaded eclipse as is, same for orientdb and jdk (I don't use maven). Then I set the latter as default jdk in eclipse config. Finally I created a new java project, including the external jars orient-core, orient-client and orient-enterprise from orientdb lib, and got that error.
If I add every jar into orientdb lib to the project everything goes fine. So I suppose there should be some other jars needed, over the three I've mentioned before.
